### DISCUSSION TITLE
Binary sent is not the same as binary received

### DISCUSSION THREAD

**Discussion by soren-olesen** [emqx/MQTTX#1692] [Q&A] [answered]
I subscribe with:
`mqttx sub -h MY_IP -p 59883 -u username -P password -t test --file-save output.dat --format binary`

I publish with:
`mqttx pub -h MY_IP -p 59883 -u username -P password -t test --file-read input.aac --format binary`

The result is that the output.dat is not the same file as the input.aac file.
Input.aac is 26KB
Output.dat is 46KB

The first number of bytes look ok (they are "text"), but everything else is wrong.

What am I doing wrong ?

**Comment by ysfscream** [comment, marked as answer]
Hi @soren-olesen

I appreciate your patience. The issue lies with the `file-read` and `format` options when publishing the message. The data read from the file is already in binary format, and the conversion process was doing unnecessary transformations, which caused the discrepancy.

To resolve this, omit the `--format` option when using the `pub` command. This will prevent the unnecessary conversion and maintain the integrity of the data. I have tested this, and the input and output files are now completely identical.

<img width="437" alt="image" src="https://github.com/emqx/MQTTX/assets/21299158/9c3fbc60-5387-489c-8c24-2e839ec682e0">

<img width="614" alt="image" src="https://github.com/emqx/MQTTX/assets/21299158/a1a720f6-e9e8-4070-9b6d-4fb0319be971">


Thank you for bringing this to our attention. We will fix it soon. 🙏

Best regards,
Shifan

**Reply by ysfscream** [reply]
The issue has been fixed in https://github.com/emqx/MQTTX/releases/tag/v1.10.1. Please update and use it. Thank you.
